tests/subsys/net/lib/nrf_cloud/fota_common/src/main.c

```diff
@@ -0,0 +1,716 @@
+		    (fota_type == NRF_CLOUD_FOTA_MODEM_DELTA)) {
+			zassert_true(nrf_cloud_fota_is_type_modem(fota_type),
+				     "Modem FOTA type not detected");
+		} else {
+			zassert_false(nrf_cloud_fota_is_type_modem(fota_type),
+				      "Non-modem FOTA type detected as modem type");
+		}
+	}
+}
+
+#define APP_FOTA_ASSERT_STR "APP FOTA enabled should correspond to the value of: " \
+			    "CONFIG_BOOTLOADER_MCUBOOT"
+
+#define BOOT_FOTA_ASSERT_STR "BOOT FOTA enabled should correspond to the value of: " \
+			     "CONFIG_BOOTLOADER_MCUBOOT + CONFIG_BUILD_S1_VARIANT + " \
+			     "CONFIG_SECURE_BOOT"
+
+#define MDM_DELTA_ASSERT_STR "DELTA MODEM FOTA enabled should correspond to the value of: " \
+			     "CONFIG_NRF_MODEM"
+
+#define FMFU_ASSERT_STR "FULL MODEM FOTA enabled should correspond to the value of: " \
+			"CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE + CONFIG_NRF_MODEM"
+
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_fota_is_type_enabled)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_FOTA_DOWNLOAD);
+
+	bool ret;
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_APPLICATION);
+	zassert_equal(IS_ENABLED(CONFIG_BOOTLOADER_MCUBOOT), ret, APP_FOTA_ASSERT_STR);
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_BOOTLOADER);
+	zassert_equal(IS_ENABLED(CONFIG_BOOTLOADER_MCUBOOT) &&
+		      IS_ENABLED(CONFIG_BUILD_S1_VARIANT) &&
+		      IS_ENABLED(CONFIG_SECURE_BOOT),
+		      ret,
+		      BOOT_FOTA_ASSERT_STR);
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_MODEM_DELTA);
+	zassert_equal(IS_ENABLED(CONFIG_NRF_MODEM), ret, MDM_DELTA_ASSERT_STR);
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_MODEM_FULL);
+	zassert_equal(IS_ENABLED(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE) &&
+		      IS_ENABLED(CONFIG_NRF_MODEM),
+		      ret,
+		      FMFU_ASSERT_STR);
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_TYPE__INVALID);
+	zassert_false(ret, "return should be false when invalid FOTA type is provided");
+}
+
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_bootloader_fota_slot_set_get_s0_get_fail)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_FOTA_DOWNLOAD);
+	struct nrf_cloud_settings_fota_job job;
+
+	job.bl_flags = NRF_CLOUD_FOTA_BL_STATUS_CLEAR;
+	job.type = NRF_CLOUD_FOTA_BOOTLOADER;
+
+	fota_download_s0_active_get_fake.custom_fake = fake_fota_download_s0_active_get__fails;
+
+	int ret = nrf_cloud_bootloader_fota_slot_set(&job);
+
+	zassert_not_equal(0, ret,
+			  "return should not be 0 when fota_download_s0_active_get fails");
+	zassert_equal(NRF_CLOUD_FOTA_BL_STATUS_CLEAR, job.bl_flags,
+		      "bl_flags should be NRF_CLOUD_FOTA_BL_STATUS_CLEAR");
+}
+
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_bootloader_fota_slot_fail)
+{
+	int ret = nrf_cloud_bootloader_fota_slot_set(NULL);
+
+	zassert_equal(-EINVAL, ret, "return should be -EINVAL when job is NULL");
+}
+
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_bootloader_fota_slot_fota_not_enabled)
+{
+	Z_TEST_SKIP_IFDEF(CONFIG_FOTA_DOWNLOAD);
```

Comment:
I could be misremembering, but I've run into troubles in the past where skipping a test marks the test suite as a failure. Is that a risk with Z_TEST_SKIP_IFDEF?

I think I only ran into that with Unity/Cmock. I think it might be different for ZTEST